types/react-test-renderer/index.d.ts

```diff
@@ -53,6 +53,20 @@ export interface TestRendererOptions {
 }
 export function create(nextElement: ReactElement, options?: TestRendererOptions): ReactTestRenderer;
 
+/**
+ * Wrap any code rendering and triggering updates to your components into `act()` calls.
+ *
+ * Ensures that the behavior in your tests matches what happens in the browser
+ * more closely by executing pending `useEffect`s before returning. This also
+ * reduces the amount of re-renders done.
+ *
+ * @param callback An asynchronous, void callback that will execute as a single, complete React commit.
+ *
+ * @see https://reactjs.org/blog/2019/02/06/react-v16.8.0.html#testing-hooks
+ */
+// the "void | undefined" is here to forbid any sneaky return values
+// tslint:disable-next-line: void-return
+export function act(callback: () => Promise<void | undefined>): Promise<undefined>;
```

Comment:
`() => Promise<void>` would essentially accept `Promise<any>`, so it's `|`ed with `undefined` to enforce no values are returned. Just `Promise<undefined>` would reject `void` functions.

Reply:
Does this affect the test renderer runtime at all if the Promise resolves to a defined value? I think some developers like to combine early returns with an evaluation (still not caring about the return value)

```ts
if (somethingThatMakesMeReturnEarly) {
  return doSomethingElse()
}
```

Reply:
What's the problem with `Promise<any>`?

I'm incurring into an error when trying to do something like this:
```js
await act(() =>
  expect(result.current.validate()).rejects.toEqual({ a: "error" })
);
```
`expect(somePromise).rejects.toEqual()` will return the same type returned by `somePromise`. Test passes, but TypeScript is complaining: `Type 'Promise<SomePromiseReturn>' is not assignable to type 'void'.`

Reply:
@diegoazh Please open an issue with the full error message. May also be an issue with the jest typings. 

In any case: I would keep it simply an just `await` `result.current.validate()` `toThrow` and wrap it in a callback that's just `void`. It's not obvious anyway that `expect.rejects` returns a promise which makes this `act` harder to read than necessary.